packages/gestalt/src/TableOfContents.js

```diff
@@ -0,0 +1,45 @@
+// @flow strict
+import { type Node } from 'react';
+import Box from './Box.js';
+import Heading from './Heading.js';
+import styles from './TableOfContents.css';
+import TableOfContentsItem from './TableOfContentsItem.js';
+
+type Props = {|
+  /**
+   * Prop description.
+   */
+  accessibilityLabel?: string,
+  title: string,
+  children: Node,
+|};
+
+/**
+ * [TableOfContents](https://gestalt.pinterest.systems/web/tableofcontents) component should be used for ... on the page.
+ * ![TableOfContents light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/TableOfContents.spec.mjs-snapshots/TableOfContents-chromium-darwin.png)
+ * ![TableOfContents dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/TableOfContents-dark.spec.mjs-snapshots/TableOfContents-dark-chromium-darwin.png)
+ */
+export default function TableOfContents({ accessibilityLabel, title, children }: Props): Node {
+  return (
+    <Box
```

Comment:
dont forget the role="navigation" check gestalt docs, it should have it
<img width="261" alt="Screenshot 2023-06-23 at 3 03 09 PM" src="https://github.com/pinterest/gestalt/assets/10593890/ea38290f-3379-49af-805d-e22fe6114600">